mcstatus/motd.py

```diff
@@ -0,0 +1,545 @@
+
+    .. seealso:: `Minecraft's wiki. <https://minecraft.fandom.com/wiki/Raw_JSON_text_format#Translated_Text>`_
+    """
+
+    id: str
+
+
+ParsedMotdItem: TypeAlias = Union[Formatting, MinecraftColor, WebColor, TranslateString, str]
+
+
+@dataclass
+class Motd:
+    """Represents parsed MOTD.
+
+    Basing on this class, you can write your own MOTD-to-something parser, see ``parsed`` field.
+
+    :param parsed: Parsed MOTD, which then will be transformed.
+        Bases on this field, you can easily write your own MOTD-to-something parser.
+    :param bedrock: Is server Bedrock Edition? Some details may change in work of this class.
+    """
+
+    parsed: list[ParsedMotdItem]
+    bedrock: bool = False
+
+    @classmethod
+    def parse(cls, raw: Union[dict, list, str], *, bedrock: bool = False) -> Self:
+        """Parse a raw MOTD to less raw MOTD (``Motd.parsed`` field).
+
+        :param raw: Raw MOTD, directly from server.
+        :param bedrock: Is server Bedrock Edition? Nothing changes here, just sets attribute.
+        :returns: New ``MOTD`` instance.
+        """
+        if isinstance(raw, list):
+            raw = {"extra": raw}
+
+        if isinstance(raw, str):
+            parsed = cls._parse_as_str(raw, bedrock=bedrock)
+        else:
+            parsed = cls._parse_as_dict(raw, bedrock=bedrock)
+
+        return cls(parsed, bedrock)
+
+    @classmethod
+    def _parse_as_dict(
+        cls, item: dict, *, bedrock: bool = False, auto_add: Optional[list[ParsedMotdItem]] = None
+    ) -> list[ParsedMotdItem]:
+        """Parse a MOTD when it's dict.
+
+        :param item: ``dict`` directly from the server.
+        :param auto_add: Values to add on this item.
+            Most time, this is ``Formatting`` from top level.
+        :returns: ``ParsedMotdItem`` list, which need to be passed to ``__init__``.
+        """
+        parsed_motd: list[ParsedMotdItem] = auto_add if auto_add is not None else []
+
+        if item.get("color"):
+            try:
+                parsed_motd.append(MinecraftColor[item["color"].upper()])
+            except KeyError:
+                if item["color"] == "reset":
+                    # minecraft server actually can't return {"reset": True}, instead
+                    # it will return {"color": "reset"}. but for follow naming, we
+                    # will not move `Formatting.RESET` to `MinecraftColor`.
+                    #
+                    # see https://wiki.vg/Chat#Shared_between_all_components, `color` field
+                    parsed_motd.append(Formatting.RESET)
+                else:
+                    parsed_motd.append(WebColor.from_hex(item["color"]))
+
+        for style_key, style_val in Formatting.__members__.items():
+            lowered_style_key = style_key.lower()
+            if item.get(lowered_style_key) is False or item.get(lowered_style_key) == "false":
+                parsed_motd.remove(style_val)
```

Comment:
Annoyingly, some servers set the formatting keys to false here, even without it ever being set to true before, which then causes a `ValueError` here as there's nothing to remove.
![image](https://user-images.githubusercontent.com/20902250/193406839-905bf7d0-8318-4ddf-b585-817b2097773e.png)

Reply:
Fixed in [last force push](https://github.com/py-mine/mcstatus/compare/ae2dd4eff4d605ae236fba6a2f87e783aa51de2e..8b3b7ad411ed3bd1ae4af2d46cf734040ac223f3), thanks.